GVFS/GVFS.Common/NuGetUpgrade/NuGetFeed.cs

```diff
@@ -168,6 +161,15 @@ private static PackageSourceCredential BuildCredentialsFromPAT(string personalAc
                 storePasswordInClearText: false);
         }
 
+        private void SetSourceRepository()
+        {
+            this.sourceRepository = Repository.Factory.GetCoreV3(this.feedUrl);

```

Comment:
Are there any affects of resetting the `sourceRepository` on the retry when we set the credentials?  Why could we not just set the `this.sourceRepository.PackageSource.Credentials` on the retry?  Is the connection bad after the failed attempt?

Reply:
> Why could we not just set the this.sourceRepository.PackageSource.Credentials on the retry?

I tried this originally, but it did not appear to actual result in the new credentials being used. I have it on my list to dig into further.

Reply:
Seems like, if anything, it's safer to dump the whole `this.sourceRepository` to prevent more subtle bugs from cropping up.